scene/gui/line_edit.cpp

```diff
@@ -503,7 +503,7 @@ void LineEdit::gui_input(const Ref<InputEvent> &p_event) {
 						last_dblclk_pos = b->get_position();
 						PackedInt32Array words = TS->shaped_text_get_word_breaks(text_rid);
 						for (int i = 0; i < words.size(); i = i + 2) {
-							if ((words[i] < caret_column && words[i + 1] > caret_column) || (i == words.size() - 2 && caret_column == words[i + 1])) {
+							if ((words[i] <= caret_column && words[i + 1] >= caret_column) || (i == words.size() - 2 && caret_column == words[i + 1])) {
```

Comment:
```suggestion
							if (words[i] <= caret_column && words[i + 1] >= caret_column) {
```
The second half of the condition is no longer needed.